### PR TITLE
koord-manager: add quota profile controller

### PR DESCRIPTION
**Pull request by shaloulcy**
### Ⅰ. Describe what this PR does

add quota profile controller. It will update the quota

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: pkg/quota-controller/profile/profile_controller.go [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+		if !reflect.DeepEqual(quota.Labels, oldQuota.Labels) || !reflect.DeepEqual(quota.Spec, oldQuota.Spec) {
+			err = r.Client.Update(context.TODO(), quota)
+			if err != nil {
+				klog.Errorf("failed update quota for profile %v, error: %v", req.NamespacedName, err)
+				return ctrl.Result{Requeue: true}, err
+			}
+		}
+	}
+
+	return ctrl.Result{Requeue: true, RequeueAfter: 10 * time.Second}, nil
```
Comment (hormes) [inline pkg/quota-controller/profile/profile_controller.go:158]: Why keep retrying?

File: pkg/quota-controller/profile/profile_controller.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+	min := corev1.ResourceList{}
+	for _, key := range resourceKeys {
+		resourceName := corev1.ResourceName(key)
+		quantity, ok := totalResource[resourceName]
+		if ok {
+			min[resourceName] = quantity
+		}
+	}
+
+	quota.Spec.Min = min
```
Comment (hormes) [inline pkg/quota-controller/profile/profile_controller.go:125]: From Sum(node Allocatable) ---> min , we need a fragmentation factor here, and this factor is configurable.

**Comment by shaloulcy** [comment]
#1606

**Pull request merged**